addons/website/models/res_company.py

```diff
@@ -19,6 +20,18 @@ def action_open_website_theme_selector(self):
         action['target'] = 'new'
         return action
 
+    @api.constrains('active')
+    def _check_active(self):
+        super()._check_active()
+        for company in self:
+            if not company.active and company.website_id:
+                raise ValidationError(_(
+                    'The company %(company_name)r cannot be archived because it has a linked website %(website_name)r.'
```

Comment:
(I already changed it so I can't comment the old code, but here is the diff)
```diff
-                    'The company %(company_name)s cannot be archived because it has a linked website %(website_name)s.'
+                    'The company %(company_name)r cannot be archived because it has a linked website %(website_name)r.'
```
Generally a quick win regarding UX/UI:
```
The company Company cannot be archived because it has a linked website Website 1.
=>
The company 'Company' cannot be archived because it has a linked website 'Website 1'.
```